plugins/linux-pulseaudio/pulse-input.c

```diff
@@ -483,13 +490,28 @@ static void pulse_destroy(void *vptr)
 /**
  * Update the input settings
  */
-static void pulse_update(void *vptr, obs_data_t *settings)
+static void pulse_update(void *vptr, obs_data_t *settings, bool input)
 {
 	PULSE_DATA(vptr);
 	bool restart = false;
 	const char *new_device;
 
 	new_device = obs_data_get_string(settings, "device_id");
+
+	if (strcmp("default", new_device) == 0) {
+		struct pulse_devices *d = bzalloc(sizeof(struct pulse_devices));
+
+		pulse_get_server_info((pa_server_info_cb_t) pulse_default_devices, (void *) d);
+
+		if (input) {
+			new_device = d->default_source_name;
+		} else {
+			new_device = d->default_sink_name;
+		}
+
+		bfree(d);
```

Comment:
You did not free the default_source_name and default_sink_name which are (should be) allocated in the callback. This will create a memory leak.